mbuild/lattice.py

```diff
@@ -165,203 +170,186 @@ def _sanitize_inputs(self, lattice_vectors, dimension,
-                                     'handed system.' .format(det))
+                                 .format(np.shape(lattice_vectors),
+                                         (self.dimension, self.dimension)))
+
+            det = np.linalg.det(lattice_vectors)
+            if abs(det) == 0.0:
+                raise ValueError('Co-linear vectors: {}'
+                                 'have a determinant of 0.0. Does not '
+                                 'define a unit cell.'
+                                 .format(lattice_vectors))
+
+            if det <= 0.0:
+                raise ValueError('Negative Determinant: the determinant '
+                                 'of {} is negative, indicating a left-'
+                                 'handed system.' .format(det))
         self.lattice_vectors = lattice_vectors
 
-    def _validate_basis_atoms(self, basis_atoms, dimension):
-        if basis_atoms is None:
-            basis_atoms = {}
-            basis_atoms = {'default': [[0. for x in range(dimension)]]}
-        elif isinstance(basis_atoms, dict):
+    def _validate_lattice_points(self, lattice_points):
+        if lattice_points is None:
+            lattice_points = {}
+            lattice_points = {'id': [[0. for x in range(self.dimension)]]}
+        elif isinstance(lattice_points, dict):
             pass
         else:
-            raise TypeError('Incorrect type, basis_atoms is of type {}, '
-                            'Expected dict.'.format(type(basis_atoms)))
+            raise TypeError('Incorrect type, lattice_points is of type {}, '
+                            'Expected dict.'.format(type(lattice_points)))
 
-        for name in basis_atoms.keys():
-            positions = basis_atoms[name]
+        for name in lattice_points.keys():
+            positions = lattice_points[name]
             for pos in positions:
-                location_check = []
-                if len(pos) != dimension:
-                    raise ValueError("Incorrect basis atom position size. "
-                                     "Basis atom {} was passed with location "
+                if len(pos) != self.dimension:
+                    raise ValueError("Incorrect lattice point position size. "
+                                     "lattice point {} has location "
                                      "{}, which is inconsistent with the "
                                      "dimension {}.".format(name, pos,
-                                                            dimension))
+                                                            self.dimension))
                 if pos is None:
                     raise ValueError("NoneType passed, expected float. "
                                      "None was passed in as position for {}."
                                      .format(name))
+                for coord in pos:
+                    try:
+                            coord = float(coord)
```

Comment:
Stray indent

Reply:
I would also probably just remove this check. You will get a `cannot compare coord to float` error down below where you check if coord is between 0 and 1.